packages/blockchain-sdk-solana/src/wallet-adapter/node-dialect-solana-wallet-adapter.ts

```diff
@@ -38,16 +44,22 @@ export class NodeDialectSolanaWalletAdapter
     }
   }
 
-  async signTransaction(tx: Transaction): Promise<Transaction> {
-    tx.partialSign(this.keypair);
-    return tx;
+  async signTransaction<T extends Transaction | VersionedTransaction>(
+    transaction: T,
+  ): Promise<T> {
+    const signer: Signer = {
+      publicKey: this.keypair.publicKey,
+      secretKey: this.keypair.secretKey,
+    };
+    const signers: Signer[] = [signer];
+    (transaction as VersionedTransaction).sign(signers);
```

Comment:
unsafe cast

Reply:
Why do you think it's unsafe? You can see the definition from the function, there are type types available and here I am trying to make sure about its type. What is your suggestion?

Reply:
does code compile w/o cast?